theme.schema.json

```diff
@@ -0,0 +1,286 @@
+        "monospace": { "type": "string", "description": "Font specifically used by monospace tags" }
+      }
+    },
+
+    "colors": {
+      "type": "object",
+      "description": "List of colours available for the theme",
+      "required": [
+        "grey-0",
+        "grey-05",
+        "grey-10",
+        "grey-20",
+        "grey-30",
+        "grey-40",
+        "grey-50",
+        "grey-60",
+        "grey-70",
+        "grey-80",
+        "grey-90",
+        "grey-100",
+        "text",
+        "brand"
+      ],
+      "patternProperties": {
+        ".*": {
+          "type": "string",
+          "pattern": "^#[0-9A-Fa-f]{6}$",
+          "description": "Hex code colour"
+        }
+      }
+    },
+
+    "fontWeights": {
+      "type": "object",
+      "properties": {
+        "base": { "$ref": "#/definitions/FontWeight" },
+        "light": { "$ref": "#/definitions/FontWeight" },
+        "medium": { "$ref": "#/definitions/FontWeight" },
+        "bold": { "$ref": "#/definitions/FontWeight" }
+      }
+    },
+
+    "radii": {
+      "type": "object",
+      "properties": {
+        "sm": { "$ref": "#/definitions/Radii" },
+        "md": { "$ref": "#/definitions/Radii" },
+        "lg": { "$ref": "#/definitions/Radii" },
+        "xl": { "$ref": "#/definitions/Radii" },
+        "round": { "enum": [ "50%", "0" ] }
+      }
+    },
+
+
+    "letterSpacings": {
+      "type": "object",
+      "properties": {
+        "base": { "type": "number" },
+        "wide": { "type": "number" },
+        "narrow": { "type": "number" }
+      }
+    },
+
+
+    "lineHeights": {
+      "type": "object",
+      "properties": {
+        "base": { "type": "number" },
+        "lg": { "type": "number" }
+      }
+    },
+
+    "shadows": {
+      "low": { "$ref": "#/definitions/BoxShadow" },
+      "medium": { "$ref": "#/definitions/BoxShadow" },
+      "high": { "$ref": "#/definitions/BoxShadow" }
+    },
+
+    "fontSizes": {
+      "$ref": "#/definitions/TshirtSizes"
+    },
+
+    "space": {
+      "$ref": "#/definitions/TshirtSizes"
+    },
+
+    "sizes": {
+      "type": "object",
+      "requied": [ "grid" ],
+      "properties": {
+        "grid": {
+          "type": "object",
+          "required": [ "lg", "md", "sm" ],
+          "additionalProperties": false,
+          "properties": {
+            "lg-docs": { "type": "string" },
+            "md-docs": { "type": "string" },
+            "sm-docs": { "type": "string" },
+            "lg": { "$ref": "#/definitions/Indices/twelve" },
+            "md": { "$ref": "#/definitions/Indices/twelve" },
+            "sm": { "$ref": "#/definitions/Indices/four" }
+          }
+        }
+      }
+    },
+
+    "zIndices": {
+      "type": "object",
+      "properties": {
+        "level": { "$ref": "#/definitions/Indices/twelve" },
+        "max": {
+          "type": "integer",
+          "minimum": 999
```

Comment:
Why minimum 999?

Reply:
So that's it's greater than the other levels and we can use the schema to help enforce maximum level z indices (although sparingly)